packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AbstractModal from 'terra-abstract-modal';
+import Button from 'terra-button';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './NotificationDialog.scss';
+
+const cx = classNames.bind(styles);
+
+const variants = {
+  ALERT: 'alert',
+  ERROR: 'error',
+  WARNING: 'warning',
+  INFO: 'info',
+  SUCCESS: 'success',
+  CUSTOM: 'custom',
+};
+
+const propTypes = {
+  /**
+   * Header of the notification-dialog.
+   */
+  header: PropTypes.string,
+  /**
+   * Title of the notification-dialog.
+   */
+  title: PropTypes.string,
+  /**
+   * Message of the notification-dialog.
+   */
+  message: PropTypes.string,
+  /**
+   * The Action of the primary button.
+   */
+  primaryAction: PropTypes.shape({
```

Comment:
Should the primary action be required? or what is the behavior if neither a primary or secondary is provided? Is it close on esc?